src/main/java/com/microsoft/azure/functions/worker/broker/JavaFunctionBroker.java

```diff
@@ -87,14 +91,48 @@ private void addSearchPathsToClassLoader(FunctionMethodDescriptor function) thro
 		function.getLibDirectory().ifPresent(d -> registerWithClassLoaderProvider(d));
 	}
 
-	private void registerWithClassLoaderProvider(File libDirectory) {
+	void registerWithClassLoaderProvider(File libDirectory) {
 		try {
-			classLoaderProvider.addDirectory(libDirectory);
+			if(SystemUtils.IS_JAVA_1_8) {
+				String workerLibPath = System.getenv(Constants.FUNCTIONS_WORKER_DIRECTORY) + "/lib";
+				String javaReverseLibLoading = System.getenv(Constants.JAVA_REVERSE_LIB_LOADING);
+				File workerLib = new File(workerLibPath);
+				if(javaReverseLibLoading != null && javaReverseLibLoading.toLowerCase().equals("true")) {
+					if(checkLibFolder(workerLib, workerLibPath, "CLIENT_LIB_FIRST")) {
+						classLoaderProvider.addDirectory(libDirectory);
+						classLoaderProvider.addDirectory(workerLib);
+					}
+				} else {
+					if(checkLibFolder(workerLib, workerLibPath, "WORKER_LIB_FIRST")) {
+						classLoaderProvider.addDirectory(workerLib);
+						classLoaderProvider.addDirectory(libDirectory);
+					}
+				}
+			} else {
+				classLoaderProvider.addDirectory(libDirectory);
+			}
 		} catch (Exception ex) {
 			ExceptionUtils.rethrow(ex);
 		}
 	}
 
+	boolean checkLibFolder(File workerLib, String workerLibPath, String status) throws Exception{
+		if(!workerLib.exists()) {
```

Comment:
`status` is only used for logging, remove this is input parameter and handle logging in the if..else above.

Reply:
You have a point the reason is as the exception is created in the method, this method now serve two purposes, exception and exception message. To change the message there is three ways, the one proposed.
second return just false and pass the excptionString function parameter and create the exception in the else.What do you think?

Reply:
This is mostly a Nit. Up to you.